### ISSUE TITLE
[Vue warn]: injection "usehead" not found. 

### ISSUE THREAD

**Issue by Nks** [danielroe/nuxt-vitest#149]
When the page has the `useHead` inside the component it won't inject it:

Page:
```
<template>
  <div>Index page</div>
</template>

<script setup>
useHead({
  title: 'test'
})

definePageMeta({
  value: 'set in index',
})
</script>
```

Test:
```
import { describe, it, expect } from 'vitest'

import { mountSuspended} from 'vitest-environment-nuxt/utils'
import Page from '~/pages/index.vue'

describe('test pages', () => {
  it('can mount the page', async () => {
    const component = await mountSuspended(Page)
    expect(component.html()).toMatchInlineSnapshot(`"<div>Index page</div>"`)
  })
})
```

The test is passed but with the warnings:

```
[Vue warn]: injection "usehead" not found. 
  at <Index> 
  at <Anonymous ref="VTU_COMPONENT" > 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <Index> 
  at <Anonymous ref="VTU_COMPONENT" > 
  at <VTUROOT>

```

As I know that nuxt changed the import for `useHead` from `#head` to `#import`, and maybe that is the case that it is not imported properly.

I also tried to use `import {mount} from "@vue/test-utils";` but it returned the same error.

Any thoughts?

**Comment by danielroe** [comment]
I believe this should be resolved in 0.6.12 with https://github.com/danielroe/nuxt-vitest/pull/147 - would you check and provide a reproduction if not?

**Comment by Nks** [comment]
Thanks for the quick update @danielroe!

I ran the test in the playground, and it returned the same warning even though the test was passed:

```
stderr | tests/nuxt/index.spec.ts > test utils > can mount components within nuxt suspense
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
```

Here is my environment:

```
$ pnpm -v
8.4.0

$ node -v                      
v16.13.2

npm -v               
8.1.2
```

Running the tests this way:
```➜  nuxt-vitest git:(main) ✗ pnpm i 
Scope: all 4 workspace projects
Lockfile is up to date, resolution step is skipped
Already up to date
. prepare$ husky install
│ husky - Git hooks installed
└─ Done in 70ms
Done in 587ms

 pnpm i
Scope: all 4 workspace projects
.. prepare$ husky install
│ husky - Git hooks installed
└─ Done in 67ms
Done in 576ms
➜  playground git:(main) ✗ pnpm test     

> @0.6.12 test /Users/nks/Projects/javascript/nuxt-vitest/playground
> pnpm test:unit --run && pnpm test:dev


> @0.6.12 test:unit /Users/nks/Projects/javascript/nuxt-vitest/playground
> vitest "--run"

From custom module!

 RUN  v0.30.1 /Users/nks/Projects/javascript/nuxt-vitest/playground

stdout | unknown test
<Suspense> is an experimental feature and its API will likely change.

stdout | unknown test
<Suspense> is an experimental feature and its API will likely change.

stdout | unknown test
<Suspense> is an experimental feature and its API will likely change.

stdout | unknown test
<Suspense> is an experimental feature and its API will likely change.

stdout | unknown test
<Suspense> is an experimental feature and its API will likely change.

stdout | unknown test
<Suspense> is an experimental feature and its API will likely change.

stdout | unknown test
<Suspense> is an experimental feature and its API will likely change.

stderr | tests/nuxt/mock-component-2.spec.ts > should mock
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>

stderr | tests/nuxt/mock-component-1.spec.ts > should mock
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>

stderr | tests/nuxt/index.spec.ts > test utils > can mount components within nuxt suspense
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>

stderr | tests/nuxt/index.spec.ts > test utils > can mount index page within nuxt suspense
[Vue warn]: injection "usehead" not found. 
  at <Index> 
  at <Anonymous ref="VTU_COMPONENT" > 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <Index> 
  at <Anonymous ref="VTU_COMPONENT" > 
  at <VTUROOT>

 ✓ tests/unit/index.spec.ts (1)
 ✓ tests/nuxt/auto-import.spec.ts (1)
 ✓ tests/nuxt/config.spec.ts (1)
 ✓ tests/nuxt/mock-component-2.spec.ts (1)
 ✓ tests/nuxt/mock-nuxt-composable.spec.ts (1)
 ✓ tests/nuxt/auto-import-mock.spec.ts (1)
 ✓ tests/nuxt/mock-component-1.spec.ts (1)
 ✓ tests/nuxt/index.spec.ts (8)

 Test Files  8 passed (8)
      Tests  13 passed | 2 todo (15)
   Start at  17:34:27
   Duration  1.99s (transform 278ms, setup 1ms, collect 798ms, tests 82ms, environment 10.30s, prepare 683ms)


> @0.6.12 test:dev /Users/nks/Projects/javascript/nuxt-vitest/playground
> NUXT_VITEST_DEV_TEST=true nuxt dev

                                                                                                                                                                                              5:34:29 PM
Nuxi 3.4.3                                                                                                                                                                                    5:34:29 PM
Nuxt 3.4.3 with Nitro 2.3.3                                                                                                                                                                   5:34:29 PM
[get-port] Unable to find an available port (tried 3000). Using alternative port: 3001                                                                                                        5:34:29 PM
                                                                                                                                                                                              5:34:30 PM
  > Local:    http://localhost:3001/ 
  > Network:  http://192.168.88.107:3001/
  > Network:  http://[fd77:53c9:ff30:42cb:14c4:7cc4:ecb0:7a7e]:3001/

✔ Nuxt Devtools is enabled v0.4.5 (experimental)                                                                                                                                              5:34:30 PM
From custom module!                                                                                                                                                                           5:34:30 PM
 RUN  v0.30.1 /Users/nks/Projects/javascript/nuxt-vitest/playground                                                                                                                           5:34:31 PM
ℹ Vite client warmed up in 766ms                                                                                                                                                              5:34:31 PM
                                                                                                                                                                                              5:34:31 PM
✓ tests/unit/index.spec.ts (1)                                                                                                                                                                5:34:31 PM
                                                                                                                                                                                              5:34:31 PM
                                                                                                                                                                                              5:34:31 PM
                                                                                                                                                                                              5:34:31 PM
                                                                                                                                                                                              5:34:31 PM
                                                                                                                                                                                              5:34:31 PM
                                                                                                                                                                                              5:34:31 PM
✔ Nitro built in 459 ms                                                                                                                                                                 nitro 5:34:32 PM
                                                                                                                                                                                              5:34:33 PM
stderr | unknown test                                                                                                                                                                         5:34:33 PM
[Vue warn]: Failed to mount app: mount target selector "#__nuxt" returned null.                                                                                                               5:34:29 PM
                                                                                                                                                                                              5:34:33 PM
stderr | unknown test                                                                                                                                                                         5:34:33 PM
[Vue warn]: Failed to mount app: mount target selector "#__nuxt" returned null.
                                                                                                                                                                                              5:34:33 PM
stderr | unknown test                                                                                                                                                                         5:34:33 PM
[Vue warn]: Failed to mount app: mount target selector "#__nuxt" returned null.
                                                                                                                                                                                              5:34:33 PM
stderr | unknown test                                                                                                                                                                         5:34:33 PM
[Vue warn]: Failed to mount app: mount target selector "#__nuxt" returned null.
                                                                                                                                                                                              5:34:33 PM
stderr | unknown test                                                                                                                                                                         5:34:33 PM
[Vue warn]: Failed to mount app: mount target selector "#__nuxt" returned null.
                                                                                                                                                                                              5:34:33 PM
stderr | unknown test                                                                                                                                                                         5:34:33 PM
[Vue warn]: Failed to mount app: mount target selector "#__nuxt" returned null.
                                                                                                                                                                                              5:34:33 PM
stderr | unknown test                                                                                                                                                                         5:34:33 PM
[Vue warn]: Failed to mount app: mount target selector "#__nuxt" returned null.
                                                                                                                                                                                              5:34:33 PM
✓ tests/unit/index.spec.ts (1)                                                                                                                                                                5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
 ✓ tests/unit/index.spec.ts (1)                                                                                                                                                               5:34:33 PM
 ✓ tests/nuxt/auto-import.spec.ts (1)
                                                                                                                                                                                              5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
stdout | tests/nuxt/mock-component-2.spec.ts > should mock                                                                                                                                    5:34:33 PM
<Suspense> is an experimental feature and its API will likely change.
                                                                                                                                                                                              5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
stdout | tests/nuxt/mock-component-2.spec.ts > should mock                                                                                                                                    5:34:33 PM
<empty line>
                                                                                                                                                                                              5:34:33 PM
stderr | tests/nuxt/mock-component-2.spec.ts > should mock                                                                                                                                    5:34:33 PM
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
                                                                                                                                                                                              5:34:33 PM
✓ tests/unit/index.spec.ts (1)                                                                                                                                                                5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
stdout | tests/nuxt/mock-component-1.spec.ts > should mock                                                                                                                                    5:34:33 PM
<Suspense> is an experimental feature and its API will likely change.
                                                                                                                                                                                              5:34:33 PM
stdout | tests/nuxt/mock-component-1.spec.ts > should mock                                                                                                                                    5:34:33 PM
<empty line>
                                                                                                                                                                                              5:34:33 PM
stderr | tests/nuxt/mock-component-1.spec.ts > should mock                                                                                                                                    5:34:33 PM
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
                                                                                                                                                                                              5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
stdout | tests/nuxt/index.spec.ts > test utils > can mount components within nuxt suspense                                                                                                    5:34:33 PM
<Suspense> is an experimental feature and its API will likely change.
                                                                                                                                                                                              5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
stdout | tests/nuxt/index.spec.ts > test utils > can mount components within nuxt suspense                                                                                                    5:34:33 PM
<empty line>
                                                                                                                                                                                              5:34:33 PM
stderr | tests/nuxt/index.spec.ts > test utils > can mount components within nuxt suspense                                                                                                    5:34:33 PM
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <VTUROOT>
                                                                                                                                                                                              5:34:33 PM
                                                                                                                                                                                              5:34:33 PM
stderr | tests/nuxt/index.spec.ts > test utils > can mount index page within nuxt suspense                                                                                                    5:34:33 PM
[Vue warn]: injection "usehead" not found. 
  at <Index> 
  at <Anonymous ref="VTU_COMPONENT" > 
  at <VTUROOT>
[Vue warn]: injection "usehead" not found. 
  at <Index> 
  at <Anonymous ref="VTU_COMPONENT" > 
  at <VTUROOT>
                                                                                                                                                                                              5:34:33 PM
 ✓ tests/unit/index.spec.ts (1)                                                                                                                                                               5:34:33 PM
 ✓ tests/nuxt/auto-import.spec.ts (1)
 ✓ tests/nuxt/config.spec.ts (1)
 ✓ tests/nuxt/mock-component-2.spec.ts (1)
 ✓ tests/nuxt/auto-import-mock.spec.ts (1)
 ✓ tests/nuxt/mock-component-1.spec.ts (1)
 ✓ tests/nuxt/mock-nuxt-composable.spec.ts (1)
 ✓ tests/nuxt/index.spec.ts (8)
 Test Files  8 passed (8)                                                                                                                                                                     5:34:33 PM
      Tests  13 passed | 2 todo (15)                                                                                                                                                          5:34:33 PM
   Start at  17:34:31                                                                                                                                                                         5:34:33 PM
   Duration  2.60s (transform 534ms, setup 0ms, collect 1.05s, tests 95ms, environment 11.39s, prepare 3.09s) 
```
